includes/class-wp-job-manager-post-types.php

```diff
@@ -736,19 +738,40 @@ public static function get_permalink_structure() {
 		}
 
-		$permalinks = wp_parse_args(
-			(array) get_option( 'wpjm_permalinks', array() ),
+		/**
+		 * Option `wpjm_permalink` was renamed to match other options in 1.32.0.
+		 *
+		 * Reference to the old option and support for non-standard plugin updates will be removed in 1.34.0.
+		 */
+		$legacy_permalink_settings = wp_json_encode( get_option( 'wpjm_permalink', array() ) );
+		$permalink_settings = (array) json_decode( get_option( self::PERMALINK_OPTION_NAME, $legacy_permalink_settings ), true );
+
+		// Check if migration happened on upgrade, if not, do it manually.
+		if ( ! array_key_exists( 'jobs_archive', $permalink_settings ) ) {
+			// Create entry to prevent future checks.
+			$permalink_settings['jobs_archive'] = '';
+			if ( current_theme_supports( 'job-manager-templates' ) ) {
+				// If the user has a theme that declares support, assume they are using the old archive page from the translation.
+				$permalink_settings['jobs_archive'] = _x( 'jobs', 'Post type archive slug - resave permalinks after changing this', 'wp-job-manager' );
+			}
+			update_option( self::PERMALINK_OPTION_NAME, wp_json_encode( $permalink_settings ) );
+		}
+
+		$permalinks         = wp_parse_args(
+			$permalink_settings,
 			array(
 				'job_base'      => '',
 				'category_base' => '',
 				'type_base'     => '',
+				'jobs_archive'  => '',
 			)
 		);
 
-		// Ensure rewrite slugs are set.
-		$permalinks['job_rewrite_slug']      = untrailingslashit( empty( $permalinks['job_base'] ) ? _x( 'job', 'Job permalink - resave permalinks after changing this', 'wp-job-manager' ) : $permalinks['job_base'] );
-		$permalinks['category_rewrite_slug'] = untrailingslashit( empty( $permalinks['category_base'] ) ? _x( 'job-category', 'Job category slug - resave permalinks after changing this', 'wp-job-manager' ) : $permalinks['category_base'] );
-		$permalinks['type_rewrite_slug']     = untrailingslashit( empty( $permalinks['type_base'] ) ? _x( 'job-type', 'Job type slug - resave permalinks after changing this', 'wp-job-manager' ) : $permalinks['type_base'] );
+		// Ensure rewrite slugs are set. Use legacy translation options if not.
+		$permalinks['job_rewrite_slug']          = untrailingslashit( empty( $permalinks['job_base'] ) ? _x( 'job', 'Job permalink - resave permalinks after changing this', 'wp-job-manager' ) : $permalinks['job_base'] );
+		$permalinks['category_rewrite_slug']     = untrailingslashit( empty( $permalinks['category_base'] ) ? _x( 'job-category', 'Job category slug - resave permalinks after changing this', 'wp-job-manager' ) : $permalinks['category_base'] );
+		$permalinks['type_rewrite_slug']         = untrailingslashit( empty( $permalinks['type_base'] ) ? _x( 'job-type', 'Job type slug - resave permalinks after changing this', 'wp-job-manager' ) : $permalinks['type_base'] );
+		$permalinks['jobs_archive_rewrite_slug'] = untrailingslashit( empty( $permalinks['jobs_archive'] ) ? 'job-listings' : $permalinks['jobs_archive'] );
```

Comment:
Note: I didn't translate the new default (`job-listings`) but would be open to it. I didn't want people to use translations to _change_ the slug (like old with the legacy support we're keeping with the other ones), but I can see the case for having the _default_ be language specific.

Reply:
Another option would be to do something like I do in most of my plugins, which is to pull the post type single label, and use that (since a lot of themes modify this to be something like `listings` or whatever custom one they use)

```php
	public static function get_job_post_label(){

		$job_obj      = get_post_type_object( 'job_listing' );
		$job_singular = is_object( $job_obj ) ? $job_obj->labels->singular_name : __( 'Job' );

		if( ! $job_singular ) $job_singular = __( 'Job' );

		return $job_singular;

	}
```